nm/src/nm.c

```diff
@@ -264,64 +261,36 @@ _j4status_nm_format_down_other_callback(const gchar *token, guint64 value, gcons
 static void
 _j4status_nm_device_get_addresses_ipv4(J4statusPluginContext *context, NMDevice *device, GVariantBuilder *builder)
 {
-    NMIP4Config *ip4_config;
+    NMIPConfig *ip4_config;
 
     ip4_config = nm_device_get_ip4_config(device);
     if ( ip4_config == NULL )
         return;
 
-    const GSList *address_;
-    for ( address_ = nm_ip4_config_get_addresses(ip4_config) ; address_ != NULL ; address_ = g_slist_next(address_) )
+    const GPtrArray *addresses_ = nm_ip_config_get_addresses(ip4_config);
+    guint i = 0;
```

Comment:
I usually use `gsize` for indexes but `->len` is `guint`… Your call.

But don’t init it here, it’s done in the loop right after it.